packages/amplify-provider-awscloudformation/src/iterative-deployment/iterative-rollback.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  $TSContext,
+  $TSMeta,
+  DeploymentState,
+  DeploymentStatus,
+  DeploymentStepStatus,
+  IDeploymentStateManager,
+  JSONUtilities,
+} from 'amplify-cli-core';
+import { DeploymentOp, DeploymentManager } from '.';
+import { S3 } from '../aws-utils/aws-s3';
+import { formUserAgentParam } from '../aws-utils/user-agent';
+import ora from 'ora';
+const spinner = ora('');
+
+const prevDeploymentStatus = [
+  DeploymentStepStatus.DEPLOYING,
+  DeploymentStepStatus.ROLLING_BACK,
+  DeploymentStepStatus.DEPLOYED,
+  DeploymentStepStatus.WAITING_FOR_ROLLBACK,
+];
+
+const loadDeploymentMeta = async (s3: S3, bucketName: string, metaKey: string): Promise<DeploymentOp> => {
+  const metaDeploymentContent = await s3.getStringObjectFromBucket(bucketName, metaKey);
+  if (metaDeploymentContent) {
+    return JSONUtilities.parse<DeploymentOp>(metaDeploymentContent);
+  }
+  throw new Error(`Could not load deployment meta for ${metaKey}`);
+};
+
+// https://stackoverflow.com/questions/1916218/find-the-longest-common-starting-substring-in-a-set-of-strings
+const getParentStatePath = (files: string[]) => {
+  let A = files.concat().sort();
```

Comment:
What are we concatenating? ES6 sort method is inplace and does not return a sorted array. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort